xdsl/dialects/builtin.py

```diff
@@ -153,11 +152,8 @@ def __init__(self, param: Iterable[AttributeCovT]) -> None:
 
     @classmethod
     def parse_parameter(cls, parser: AttrParser) -> tuple[AttributeCovT, ...]:
-        data = parser.parse_comma_separated_list(
-            parser.Delimiter.SQUARE, parser.parse_attribute
-        )
-        # the type system can't ensure that the elements are of type _ArrayAttrT
-        return cast(tuple[AttributeCovT, ...], tuple(data))
+        # This functionality is provided by the attribute parser.
+        raise NotImplementedError
```

Comment:
Should we have an assertion instead?
I feel `NotImplementedError` looks like something could be implemented, while this is not the case.

Reply:
If we're doing this I'd prefer both the parse and print to be not implemented.